src/jit/morph.cpp

```diff
@@ -4395,24 +4429,26 @@ void Compiler::fgMorphSystemVStructArgs(GenTreeCall* call, bool hasStructArgumen
                         // Create LCL_FLD for each eightbyte.
                         argListCreated = true;
 
+                        // First field
+                        arg->AsLclFld()->gtFieldSeq = FieldSeqStore::NotAField();
+                        arg->gtType =
+                            GetTypeFromClassificationAndSizes(fgEntryPtr->structDesc.eightByteClassifications[0],
+                                                              fgEntryPtr->structDesc.eightByteSizes[0]);
+                        GenTreeFieldList* fieldList =
+                            new (this, GT_FIELD_LIST) GenTreeFieldList(arg, 0, originalType, nullptr);
+                        fieldList->gtType          = originalType; // Preserve the type. It is a special case.
+                        arg = fieldList;
```

Comment:
GT_FIELD_LIST of GT_LCL_FLD:

I see GT_FIELD_LIST has both offset and fieldType and similar info is carried by GT_LCL_FLD.

Looks like in future we should be able to just create GT_FIELD_LIST with arg being GenTree that evaluates to a struct and offset, fieldType stored in GT_FIELD_LIST.